content/blog/kubernetes-clusters-as-code-part1.md

```diff
@@ -0,0 +1,756 @@
+---
+title: Kubernetes Cluster as Code - Part 1
+date: 2022-06-23T10:15:09.699Z
+author: Rachana Kallada Jayaraj
+authorimage: /img/photo-1-.jpg
+tags:
+  - hpe-greenlake
+  - devops
+  - terraform
+  - open source
+---
+## Getting started
+
+The process of managing and provisioning computer data centers through machine-readable definition files, also known as Infrastructure-as-Code (IaC), offers many significant benefits. It helps to increase operational agility, simplify management, reduce errors, and save cost. In this post, we will explore some of the benefits of using IaC to build a Kubernetes cluster from scratch, with all the necessary configuration and core services, on HPE GreenLake using Terraform (TF). Storing Kubernetes cluster and favorable configurations as code helps in repeatability and change management.
+
+IaC with Kubernetes is not new. There are providers in the developer community that are quite good and well supported. Using the HPE GreenLake Terraform provider, you can bring up a Kubernetes cluster starting right from the infrastructure layer and way up in the stack to set up desired configurations and applications. For reference, see the below picture.
+
+![](/img/image2022-6-20_12-36-56.png)
+
+HPE GreenLake TF provider brings the Kubernetes stack up on the HPE GreenLake Infrastructure, and exposes credentials for other TF providers to integrate further and build the complete stack, as desired. In the diagram above, 2 and 3 are community providers that are available, which can be used in combination with HPE GreenLake TF provider.
+
+In this two-part blog series, I’ll share my experience as a first-time user of HPE Greenlake TF provider. This blog series aims to provide a step by step walkthrough of how to bring up a cluster using Terraform and how to deploy applications on the pre-created cluster. 
```

Comment:
Suggest to specify you are going to create a Kubernetes cluster.

...to bring up a Kubernetes cluster using Terraform and how to deploy applications on the pre-created Kubernetes cluster.

Does it make sense to you?

Reply:
Yes, I have made this change.

Reply:
Thank you!!